syntax = "proto3";

/* Define package */
package pymerang;

/* Import status codes */
import "pymerang/commons.proto";
import "pymerang/status_codes.proto";


/* RPC methods implementing a procedure to connect a device to a controller */

service CertAuthority {
  // Request to generate a certificate
  rpc SignCertificate(SignCertificateRequest) returns (SignCertificateReply) {}
}

/* Request for a certificate */
message SignCertificateRequest {
  Device device = 1;
  bytes csr = 2;
  AuthData auth_data = 3;
}

/* Reply containing a certificate */
message SignCertificateReply {
  StatusCode status = 1;
  bytes cert = 2;
}